specs/phase1/shard-transition.md

```diff
@@ -58,16 +64,18 @@ def verify_shard_block_signature(beacon_parent_state: BeaconState,
     return bls.Verify(proposer.pubkey, signing_root, signed_block.signature)
 ```
 
-## Shard state transition
+## Shard state transition function
+
+The post-state corresponding to a pre-state `shard_state` and a signed block `signed_block` is defined as `shard_state_transition(shard_state, signed_block, beacon_parent_state)`, where `beacon_parent_state` is the parent beacon state of the `signed_block`. State transitions that trigger an unhandled exception (e.g. a failed `assert` or an out-of-range list access) are considered invalid. State transitions that cause a `uint64` overflow or underflow are also considered invalid.
 
 ```python
 def shard_state_transition(shard_state: ShardState,
                            signed_block: SignedShardBlock,
-                           validate: bool = True,
-                           beacon_parent_state: Optional[BeaconState] = None) -> ShardState:
-    if validate:
-        assert beacon_parent_state is not None
-        assert verify_shard_block_message(beacon_parent_state, shard_state, signed_block.message)
+                           beacon_parent_state: BeaconState,
+                           validate_result: bool = True) -> ShardState:
```

Comment:
```suggestion
                           validate_signature: bool = True) -> ShardState:
```

`validate_signature` sounds more appropriate?  
Also curious when we will use `validate_result=False`. Is that more for test setup?

Reply:
it's for block production. A validator initially makes the block and doesn't have the result(s) so they can't yet be validated. In the case of beaconblocks, the sig and state root (results) are then added. In the case of shard blocks, just the signature (result) is added (though we might consider putting the shard state root in there)

Reply:
Got it. Thanks!